api/tests/src/api/users/test_user_saved_opportunities_get.py

```diff
@@ -264,3 +264,27 @@ def test_user_get_only_own_saved_opportunities(
     assert "User's Opportunity" in opportunity_titles
     assert "Shared Opportunity" in opportunity_titles
     assert "Other User's Opportunity" not in opportunity_titles
+
+
+def test_user_get_saved_opportunities_deleted(
+    client, enable_factory_create, db_session, user, user_auth_token
+):
+    """Test that users don't get soft deleted opportunities"""
+    # Created a saved opportunity that is soft deleted
+    opp = OpportunityFactory.create(opportunity_title="Testing Opportunity")
+    UserSavedOpportunityFactory.create(user=user, opportunity=opp, is_deleted=True)
+
+    # Make the request
+    response = client.post(
+        f"/v1/users/{user.user_id}/saved-opportunities/list",
+        headers={"X-SGG-Token": user_auth_token},
+        json={
+            "pagination": {
+                "page_offset": 1,
+                "page_size": 25,
+            }
+        },
+    )
+
+    assert response.status_code == 200
+    assert len(response.json["data"]) == 0
```

Comment:
I'd recommend making this test have a mix of deleted and not-deleted records. Testing things separately (ie. a user has only one deleted, or only one non-deleted opportunities) doesn't always work and it's good practice to test a realistic case where a user has a mix of scenarios.